FLEDGE_implementation_overview.md

```diff
@@ -0,0 +1,64 @@
+
+Nothing is reported to the server, and the win is not recorded, until the winning ad is navigated to, or deprecatedURNToURL() is invoked, which may happen while still running reporting scripts.  If the ad loads before a reporting script completes, and tries to send a report that depends on reportWin() / reportResult(), and that function hasn’t completed, the report is queued until the script has finished running.  In general, the run duration of these should not be web visible, though if the frame the auction ran in is destroyed before they complete, any currently in-progress work will be aborted.  This is considered a bug that needs to be fixed.  Running reporting scripts can affect the performance of running auctions, since worklet processes used by reporting count towards the global Protect Audience process quotas.
+
+
+## Component Auctions
+In the case of top-level auction with multiple nested component auctions, the auctions are largely run independently, except that the interest group loading phases for all auctions must complete before any auction enters the generate bid phase, and the top-level auction only requests an executor/process for itself once all component auctions have received one, to avoid deadlock.  The top-level auction scores bids as soon as it receives winning bids from each component auction, and has its executor loaded.  Each component auction unloads the seller executor once all bids have been scored, again to avoid deadlock.  So the reporting phase may have to reload both the winning component seller worklet, and the winning bidder worklet.
+
+Timeouts are only scoped to the auction associated with the AuctionConfig that specifies them.  The top-level auctions perBuyer[Cumulative]Timeouts are ignored, and its sellerTimeout only applies to its own scoreAd() calls, not to those of component auctions.
+
+As hinted in the bid generation section, if two component auctions share a bidder, that bidder will use the same process in both auctions, reuse executors, and share a Javascript execution thread.  Worklets are run on a first-come, first-served basis (though if the interest groups of a single buyer need multiple different executors, interest groups with a ready executor and loaded signals can jump the queue), and component auctions start generateBid() calls in the order they appear in the top-level AuctionConfig, so it is recommended that the order component AuctionConfig’s are listed be randomized, if cumulative buyer timeouts are in use, since they run concurrently, and, on a per-bidder basis, the earlier an auction’s bidders queue up to run generateBid() are more likely they are to fall within the timeout.
```

Comment:
Not sure if it's worth mentioning or not, but:
Each component waits for its own promises; top-level waits for all the promises in the auction (both top-level and component).

Reply:
I think this is definitely worth mentioning!  Though the top-level auction does actually wait for all component auctions to resolve their configs as well, doesn't it?

Reply:
Yes, it needs everything, including the components (mostly because we have to serialize AuctionConfig to scoreAd)